File: Builder/A2/Director.java
public class Director {
    private Builder builder;

    public Director(Builder builder) {
        this.builder = builder;
    }

    public void construct() {
        builder.makeTitle("Bye");
        builder.makeString("気軽な別れのあいさつ");
        builder.makeItems(new String[]{
            "See you.",
            "Take care.",
            "Bye."
        });
        builder.makeString("丁寧な別れのあいさつ");
        builder.makeItems(new String[]{
            "It was nice to see you.",
            "I'm looking forward to seeing you again.",
        });
        builder.close();
    }
}

File: Builder/A2/ruby/director.rb
class Director
  def initialize(builder)
    @builder = builder
  end

  def construct
    @builder.make_title("サッカーのポジション")
    @builder.make_string("大雑把なポジション")
    @builder.make_items(["FW", "MF", "DF", "GK"])
    @builder.make_string("細かいポジション")
    @builder.make_items(["CF", "ST", "WG", "OMF", "DMF", "CMF", "SH", "SB", "CB", "GK"])
    @builder.close
  end
end
